railties/lib/rails/generators/rails/app/interactive_options.rb

```diff
@@ -0,0 +1,144 @@
+# frozen_string_literal: true
+
+module Rails
+  module Generators
+    class InteractiveOptions # :nodoc:
+      include Database
+      include Thor::Shell
+
+      attr_reader :options
+
+      def self.perform(initial_options)
+        new(initial_options).perform
+      end
+
+      def initialize(initial_options)
+        @options = initial_options
+      end
+
+      def perform
+        say "Rails CLI #{Rails.version}", :green
+
+        case inquire_rails_version
+        when "dev"    then options[:dev] = true
+        when "edge"   then options[:edge] = true
+        when "main" then options[:main] = true
+        end
+
+        options[:database]             = inquire_database
+        options[:skip_action_cable]    = inquire_action_cable
+        options[:skip_action_mailer]   = inquire_action_mailer
+        options[:skip_action_mailbox]  = inquire_action_mailbox
+        options[:skip_active_storage]  = inquire_active_storage
+        options[:skip_action_text]     = inquire_action_text
+        options[:skip_bootsnap]        = inquire_bootsnap
+        options[:skip_jbuilder]        = inquire_jbuilder
+        options[:skip_javascript]      = inquire_javascript
+        options[:skip_spring]          = inquire_spring
+        options[:skip_sprockets]       = inquire_sprockets
+        options[:skip_system_tests]    = inquire_system_tests
+        options[:skip_turbolinks]      = inquire_turbolinks
+        options[:skip_webpack_install] = inquire_webpack
+        options[:webpack]              = inquire_webpack_library unless options[:skip_webpack_install]
+
+        options
+      end
+
+      private
+        def inquire_rails_version
+          say <<~MESSAGE
+          Target different versions of rails for your application.
+          *dev* will target your checked out version of rails.
+          *edge* will target the latest commit on rails.
+          *main* will target the rails main branch.
+          *stable* will target the rails gem version you have installed.
+          MESSAGE
+          inquire_from_list("What is your preferred rails version?", %w(dev edge main stable), "stable")
+        end
+
+        def inquire_database
+          inquire_from_list("What is your preferred database?", DATABASES, options[:database])
+        end
+
+        def inquire_action_cable
+          say "Action cable integrates web sockets with your rails application: https://guides.rubyonrails.org/action_cable_overview.html"
+          inquire_boolean("Skip action cable?", options[:skip_action_cable])
```

Comment:
One thought on the skipping.

I think it makes sense when we're adding optional flags to our command e.g. `rails new . --skip-some-thing` but when we're doing an interactive experience perhaps we should have it be a positive affirmation. It feels weird having to respond in the negative to keep something.

The question could be "Include action cable?" and then toggle the skip based on the answer.

Reply:
Agreed 👍🏼